src/OrchardCore/OrchardCore.Infrastructure/Documents/VolatileDocumentManager.cs

```diff
@@ -82,5 +88,11 @@ public Task UpdateAtomicAsync(Func<Task<TDocument>> updateAsync, Func<TDocument,
 
             return Task.CompletedTask;
         }
+
+        private class UpdateDelegates
```

Comment:
That is called a `Tuple`, there are nice syntaxes in C# for that, and you can even do a struct record otherwise, but tuples are better here.

Reply:
Yes good ideas

But for the struct, I'm using a `GetOrCreateFeature()` that is waiting for a class, then for the tuple, because we can pass a factory I also tried a `Tuple.Create<UpdateDelegate, AfterUpdateDelegate>(null, null))`, but then we can't do `delegates.Item1 += () => updateAsync()` as it is readonly.